app/v2/services/ConversionService.scala

```diff
@@ -36,25 +39,37 @@ import scala.util.control.NonFatal
 @ImplementedBy(classOf[ConversionServiceImpl])
 trait ConversionService {
 
-  def convertXmlToJson(messageType: MessageType, source: Source[ByteString, _])(implicit
+  def convert(messageType: MessageType, source: Source[ByteString, _], headerType: HeaderType)(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext,
     materializer: Materializer
   ): EitherT[Future, ConversionError, Source[ByteString, _]]
 
+  def jsonToXml(messageType: MessageType, source: Source[ByteString, _])(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext,
+    materializer: Materializer
+  ): EitherT[Future, ConversionError, Source[ByteString, _]] = convert(messageType, source, HeaderTypes.jsonToXml)
+
+  def xmlToJson(messageType: MessageType, source: Source[ByteString, _])(implicit
```

Comment:
Similarly, we won't be converting from XML to JSON here, I don't think

Reply:
See above.